### PR TITLE
feat(core): create register customer page

### PR DESCRIPTION
**Pull request by yurytut1993**
## What/Why?
This pr adds Register Customer Page with Built In Fields
register-customer-form

## Testing
Locally

### CODE REVIEW

File: packages/components/src/components/form/form.tsx [hunk 1]
```diff
@@ -15,7 +15,8 @@ type ValidationPattern =
   | 'tooShort'
   | 'typeMismatch'
   | 'valid'
-  | 'valueMissing';
+  | 'valueMissing'
+  | ValidationFunction;
```
Comment (bc-alexsaiannyi) [inline packages/components/src/components/form/form.tsx:19]: I think there can be an error in `ControlValidationPatterns` so basically you can update and use `ControlValidationPatterns` after:
` type ControlValidationPatterns = ValidationPattern | ValidationFunction;`
What do you think?

Comment (yurytut1993) [reply packages/components/src/components/form/form.tsx]: Yes, changed after your updates

File: apps/core/app/[locale]/(default)/account/register-customer/page.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import { NextIntlClientProvider } from 'next-intl';
+import { getMessages, getTranslations } from 'next-intl/server';
+
+import { auth } from '~/auth';
+import { FormFieldSortInput } from '~/client/generated/graphql';
+import { getCountries } from '~/client/management/get-countries';
+import { getAddressFormFields } from '~/client/queries/get-address-form-fields';
+import { getCustomerFormFields } from '~/client/queries/get-customer-form-fields';
+import { getReCaptchaSettings } from '~/client/queries/get-recaptcha-settings';
+import { getStoreCountry } from '~/client/queries/get-store-country';
```
Comment (jorgemoya) [inline apps/core/app/[locale]/(default)/account/register-customer/page.tsx:11]: Can we do this with a single query?

File: apps/core/components/register-customer-form/_actions/login.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export const login = async (
+  email: FormDataEntryValue | null,
+  password: FormDataEntryValue | null,
+) => {
+  try {
+    const singin = await signIn('credentials', {
+      email,
+      password,
+      redirectTo: '/account',
```
Comment (jorgemoya) [inline apps/core/components/register-customer-form/_actions/login.ts:15]: I believe we already redirect to account but I might be wrong.

**Review by jorgemoya — COMMENTED** [review]
IMO `/register-customer-form` should be nested under a `_components` folder in the `register-customer` path, since it's not reused.

File: apps/core/app/[locale]/(default)/account/register-customer/page.tsx (hunk 2 above)
Comment (yurytut1993) [reply apps/core/app/[locale]/(default)/account/register-customer/page.tsx:11]: Interesting idea, but `getAddressFormFields`, `getAddressFormFields` and `getReCaptchaSettings` need to be used in separate cases 🤔

File: apps/core/components/register-customer-form/_actions/login.ts (hunk 3 above)
Comment (yurytut1993) [reply apps/core/components/register-customer-form/_actions/login.ts:15]: Yes, we have, but `submitLoginForm` action used for form and has different arguments. And using it here requires huge refactoring and can add mess in currently existed action.

File: apps/core/app/[locale]/(default)/account/register-customer/page.tsx (hunk 2 above)
Comment (jorgemoya) [reply apps/core/app/[locale]/(default)/account/register-customer/page.tsx:11]: You can try something like what @deini is doing with [fragment colocation](https://github.com/bigcommerce/catalyst/pull/766). What do you think?

File: apps/core/components/register-customer-form/_actions/login.ts (hunk 3 above)
Comment (jorgemoya) [reply apps/core/components/register-customer-form/_actions/login.ts:15]: This is okay then!

**Comment by bc-alexsaiannyi** [comment]
> IMO `/register-customer-form` should be nested under a `_components` folder in the `register-customer` path, since it's not reused.

Hmm, It's interested point. I guess that in this case it makes sense to move `contactUs` form to relevant page `_components `and delete `core/components/forms` altogether. What do you think guys?🤔

**Comment by yurytut1993** [comment]
> > IMO `/register-customer-form` should be nested under a `_components` folder in the `register-customer` path, since it's not reused.
> 
> Hmm, It's interested point. I guess that in this case it makes sense to move `contactUs` form to relevant page `_components `and delete `core/components/forms` altogether. What do you think guys?🤔

It will be consistent. And `blog-post-card` also should be replaced under `Blog` then

File: apps/core/components/register-customer-form/_actions/register-customer.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      if (sections.includes('address')) {
+        acc.address[key] = value;
+      }
+
+      return acc;
+    },
+    { address: {} },
+  );
+
+  console.log(parsedData, 'parsedData');
```
Comment (bc-alexsaiannyi) [inline apps/core/components/register-customer-form/_actions/register-customer.ts:43]: don't forget to remove console

File: apps/core/components/register-customer-form/index.tsx [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+    setCountryStatesPending(true);
+
+    const { data = [] } = await getShippingStates(country?.id || fallbackCountryId);
+
+    setCountryStatesPending(false);
+
+    setCountryStates(data);
+  };
+
+  const handlePicklistWihtTextValidation = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (bc-alexsaiannyi) [inline apps/core/components/register-customer-form/index.tsx:211]: here is a typo in method's name.

File: apps/core/app/[locale]/(default)/account/register-customer/page.tsx (hunk 2 above)
Comment (yurytut1993) [reply apps/core/app/[locale]/(default)/account/register-customer/page.tsx]: @jorgemoya thats great idea, thanks. Refactored according to the new approach 👍

File: apps/core/app/[locale]/(default)/login/register-customer/page-data.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
+import { cache } from 'react';
+
+import { getSessionCustomerId } from '~/auth';
+import { client } from '~/client';
+import { FORM_FIELDS_FRAGMENT } from '~/client/fragments/form-fields';
+import { graphql, VariablesOf } from '~/client/graphql';
+import { revalidate } from '~/client/revalidate-target';
+
+const RegisterCustomerQuery = graphql(
```
Comment (jorgemoya) [inline apps/core/app/[locale]/(default)/login/register-customer/page-data.ts:9]: I believe we can use fragment collocation better, similar to https://github.com/bigcommerce/catalyst/pull/848, but not sure if it should be part of this effort.

**Review by bc-alexsaiannyi — APPROVED** [review]
looks good to me

**Review by jorgemoya — APPROVED** [review]

**Review by bc-azvierieva — APPROVED** [review]
Nice job!

**Pull request merged**